back-end/tests/test_model.py

```diff
@@ -281,16 +281,18 @@ class TestModelUpload:
         def test_model_upload(self, client, reset_db, basedir, test_data):
             input_data = test_data["input"].copy()
 
-            if "weight_file_path" in input_data:
+            if "weight_file" in input_data:
                 weight_file_path = os.path.join(basedir, input_data["weight_file"])
-                with open(weight_file_path, "rb") as f:
-                    input_data["weight_file"] = (
-                        FileStorage(stream=f, filename="SimpleCNN.pth"),
-                    )
+                f = open(weight_file_path, "rb")
```

Comment:
Just curious: why do we not want to use `with` clause here?

Reply:
Because the `input_data` actually contains `f`. If we use `with` here and post the `input_data` outside the context like what we did earlier, the `f` will no longer be accessible.

Reply:
Just found this issue because earlier due to a typo, the body of the if statement had never been executed.